src/lib.rs

```diff
@@ -115,6 +118,51 @@ mod errors {
 }
 pub use errors::*;
 
+/// A builder for configuring a `Mapping`.
+///
+/// ```
+/// # fn foo() -> addr2line::Result<()> {
+/// use addr2line::Options;
+///
+/// let mapping = Options::default()
+///     .with_functions()
+///     .build("path/to/some/executable")?;
+/// # let _ = mapping;
+/// # }
+/// ```
```

Comment:
I'd like this to include a statement along the lines of "if `with_functions` is not added, `locate()` will always return `None` for the function".